### ISSUE TITLE
Is keyValueList non recursive anymore?

### ISSUE THREAD

**Issue by mvsmal** [fable-compiler/Fable#1597]
### Description

`kevValueList` seems to be non-recursive anymore, so if there is a property with list of DUs, it will be translated to a list, not an object. It worked recursively in Fable 1.

### Repro code

```fsharp
open Fable.Core
open Fable.Core.JsInterop
open Fable.Import.Browser

type Type1 =
    | Op1 of string
    | Op2 of string

type Type2 =
    | Op3 of Type1 list
    | Op4 of string

let foo = [ Op4 "op4"; Op3 [ Op1 "op1"; Op2 "op2" ]]

let result = keyValueList CaseRules.LowerFirst foo

console.log(result)
```

### Expected and actual results

Expected: `{"op4":"op4","op3":{"op1":"op1","op2":"op2"}}`
Actual: `{"op4":"op4","op3":[["Op1","op1"],["Op2","op2"]]}`

### Related information

* Fable version (`dotnet fable --version`): 2.0.1 / REPL2
* Operating system: Windows 10

**Comment by voronoipotato** [comment]
why would a list be represented as an object?

**Comment by mvsmal** [comment]
@voronoipotato I am not saying that this behavior is incorrect, I am just saying that in Fable 1 `keyValueList` is recursive and a property with a type of list of DUs is compiled to an object, but in Fable 2 it is compiled to a list of lists like this: `{"op3":[["Op1","op1"],["Op2":"op2"]]}`.

So it is a breaking change, which I didn't find to be documented.
My question: was it actually done on purpose?

**Comment by alfonsogarciacaro** [comment]
Good question! TBH, I don't remember exactly 😅 I think part of the reasoning was as @voronoipotato to prevent unexpected results. In Fable 1 we made `keyValueList` recursive for convenience and also because no React prop would accept `List`, as this is an F#-only type. However, in Fable 2, `keyValueList` accepts any sequence/iterable including JS arrays (the name was maintained though to make the update easier), which is problematic:

- If we keep the recursion but only for lists, users may wonder why `keyValueList` itself accepts other seq types.
- If we made the recursion available for any recursive type, it wouldn't be possible to define a prop that accepts an array.

I believe this was the main motivation to drop out recursion. But you're right, I forgot to document it, sorry! A trick to make your library match the new behavior without affecting users is converting the recursive case into a function [as we did for `Style`](https://github.com/fable-compiler/fable-react/blob/b79c43389db5a051295e2642fc9ba86f3d62aece/src/Fable.React/Fable.Helpers.React.fs#L728-L729). Would that work for you?
